internal/framework/services/s3management/resource_s3_accesskey.go

```diff
@@ -0,0 +1,232 @@
+	resp.TypeName = req.ProviderTypeName + "_s3_accesskey"
+}
+
+// Schema returns the schema for the accesskey resource.
+func (r *accesskeyResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID (UUID) of the AccessKey.",
+			},
+			"description": schema.StringAttribute{
+				Description: "Description of the Access key.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"accesskey": schema.StringAttribute{
+				Description: "Access key metadata is a string of 92 characters.",
+				Computed:    true,
+			},
+			"secretkey": schema.StringAttribute{
+				Description: "The secret key of the Access key.",
+				Computed:    true,
+			},
+			"canonical_user_id": schema.StringAttribute{
+				Description: "The canonical user ID which is valid for user-owned buckets.",
+				Computed:    true,
+			},
+			"contract_user_id": schema.StringAttribute{
+				Description: "The contract user ID which is valid for contract-owned buckets",
+				Computed:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+// Configure configures the accesskey resource.
+func (r *accesskeyResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*services.SdkBundle)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *s3.APIClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create creates the accesskey.
+func (r *accesskeyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	if r.client == nil {
+		resp.Diagnostics.AddError("s3 api client not configured", "The provider client is not configured")
+		return
+	}
+
+	var data *s3managementService.AccesskeyResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createTimeout, diags := data.Timeouts.Create(ctx, utils.DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	ctx, cancel := context.WithTimeout(ctx, createTimeout)
+	defer cancel()
+
+	var accessKey = s3management.AccessKeyCreate{
+		Properties: &s3management.AccessKeyProperties{
+			Description: data.Description.ValueStringPointer(),
+		},
+	}
+	accessKeyResponse, _, err := r.client.S3ManagementClient.CreateAccessKey(ctx, accessKey, createTimeout)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create accessKey", err.Error())
+		return
+	}
+
+	s3managementService.SetAccessKeyPropertiesToPlan(data, accessKeyResponse)
```

Comment:
do you need this?

Reply:
yes, secretkey is only returned on create and canonicalUserId only on get

Reply:
can you make that more obvious? maybe use more specific functions or adding a comment.

Reply:
sure I will do that